### PR TITLE
fix(githubPane): show issues and PR in github panes by pulling from upstream

### PR DESCRIPTION
**Pull request by kvnloo**
## Base Branch

- [x] This PR targets the `develop` branch (required for all feature/fix PRs)
- [ ] This PR targets `main` (hotfix only - maintainers)

## Description

Implement fork repository detection and dual-repository support in the GitHub integration to fix issue and PR loading failures for forked repositories. Currently, issues and PRs don't load when users work with forked repositories because these resources exist in the upstream/parent repository, not the fork itself. This feature will automatically detect forks via GitHub API, store both fork and parent repository references, and intelligently route API calls to the appropriate repository based on the operation type.

## Type of Change

- [x] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 📚 Documentation
- [ ] ♻️ Refactor
- [ ] 🧪 Test

## Area

- [ ] Frontend
- [x] Backend
- [ ] Fullstack

## Checklist

- [x] I've synced with `develop` branch
- [x] I've tested my changes locally
- [x] I've followed the code principles (SOLID, DRY, KISS)
- [x] My PR is small and focused (< 400 lines ideally)

## CI/Testing Requirements

- [x] All CI checks pass
- [ ] All existing tests pass
- [x] New features include test coverage
- [x] Bug fixes include regression tests

## Summary by CodeRabbit

* **New Features**
  * Added fork detection and a new detection channel; repo operations now route by fork status (issues/PRs to parent when appropriate, code to fork).
  * Automatic fetch-with-fallback that retries the fork repo if parent requests fail, with operation-aware routing.

* **Bug Fixes**
  * Improved repository reference normalization and parsing of quoted/empty env values; better handling of missing config fields and edge cases.

* **Tests**
  * Added comprehensive unit tests for fork-detection and fallback logic; one flaky test marked as expected-fail.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by AndyMik90 — COMMENTED** [review]
## 🤖 Auto Claude PR Review

### Merge Verdict: ✅ READY TO MERGE
No blocking issues found

### Risk Assessment
| Factor | Level | Notes |
|--------|-------|-------|
| Complexity | High | Based on lines changed |
| Security Impact | None | Based on security findings |
| Scope Coherence | Good | Based on structural review |

### Findings Summary
- **Medium**: 2 issue(s)
- **Low**: 3 issue(s)

---
_Generated by Auto Claude PR Review_

### Findings (2 selected of 5 total)

#### 🟡 [MEDIUM] Issue count query doesn't use fork-aware routing
📁 `apps/frontend/src/main/ipc-handlers/github/repository-handlers.ts:55`

The registerCheckConnection function fetches issue count using `normalizedRepo` (the fork) directly instead of using `getTargetRepo(config, 'issues')`. For forked repositories, issues exist in the parent repo, so this will return 0 or incorrect counts. The connection status will show misleading information for fork users.

**Suggested fix:**
```
Use getTargetRepo(config, 'issues') for the issue count query, similar to how issue-handlers.ts routes to the parent repo.
```

#### 🟡 [MEDIUM] githubFetchWithFallback implemented but never used
📁 `apps/frontend/src/main/ipc-handlers/github/utils.ts:193`

The githubFetchWithFallback function is fully implemented with 403/404 fallback logic and has comprehensive tests (100+ lines), but none of the handlers (issue-handlers.ts, pr-handlers.ts) actually call it. They only use getTargetRepo() for routing. This means if the parent repo returns 403 (e.g., private parent), requests will fail instead of falling back to the fork.

**Suggested fix:**
```
Either integrate githubFetchWithFallback into the handlers where fallback behavior is desired, or document that it's intended for future use.
```

---
*This review was generated by Auto Claude.*

File: apps/frontend/src/main/ipc-handlers/github/repository-handlers.ts [hunk 1]
```diff
@@ -59,15 +122,27 @@ export function registerCheckConnection(): void {
 
         const openCount = Array.isArray(issuesData) ? issuesData.length : 0;
 
+        // Build response data with fork status
```
[the remark this answers is not in the document]
Comment (AndyMik90) [reply apps/frontend/src/main/ipc-handlers/github/repository-handlers.ts:130]: @kvnloo please check if this is the case

**Comment by AndyMik90** [comment]
👋 Thanks so much for your interest in contributing to Aperant!

**Pull requests are currently paused.** We're rebuilding the entire app from the ground up for **Aperant 3.0** (including new cloud features). Most of that work is happening in a separate development repo right now and will be merged here once it's ready — so although this repo looks quiet, the project is very much alive. 🚀

Because the current codebase is being fully replaced, we're closing existing PRs so contributors aren't left waiting on changes that unfortunately can't be merged. This is **not** a reflection on your work, and we're sorry for any wasted effort. 🙏

We'd genuinely love your help shaping 3.0 instead:

- 🐛 **Report a bug:** https://github.com/AndyMik90/Aperant/issues
- 💡 **Share an idea / feature request:** https://github.com/AndyMik90/Aperant/discussions
- 💬 **Join the Discord:** https://discord.gg/KCXaPBr4Dj

Contributions will reopen once the 3.0 foundation lands. Thank you for understanding!
